### PR TITLE
feat(ctb): simplify XDM initialization

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Simplifies XDM initialization by moving all input variables into the constructor. This was the best I could do, we still need an initializer because of the stuff that the base contract inherits. The primary benefit of this change is that it means we never have to put any parameters into the initializer, which means we can be very certain about parameters before we perform the upgrade.

**Tests**
Core functionality is not changing and tests continue to pass.

### CODE REVIEW

**Comment by tynes** [comment]
I think this will also need changes in the go L1 genesis creation code

File: op-chain-ops/immutables/immutables_test.go [hunk 1]
```diff
@@ -13,6 +13,9 @@ func TestBuildOptimism(t *testing.T) {
 		"L2StandardBridge": {
 			"otherBridge": common.HexToAddress("0x1234567890123456789012345678901234567890"),
 		},
+		"L2CrossDomainMessenger": {
+			"otherMessenger": common.HexToAddress("0x1234567890123456789012345678901234567890"),
+		},
```
Comment (maurelian) [inline op-chain-ops/immutables/immutables_test.go:18]: Should these have different addresses?

Comment (smartcontracts) [reply op-chain-ops/immutables/immutables_test.go:18]: I don't think it really matters, they're just dummy values

**Comment by tynes** [comment]
```
Encountered 1 failing test in contracts/test/OptimismPortal.t.sol:OptimismPortal_FinalizeWithdrawal_Test [FAIL. Reason: EvmError: Revert Counterexample: calldata=0x2147ae39000000000000000000000000b4c79dab8f259c7aee6e5b2aa729821864227e8400000000000000000000000000000000000000000000000000000000000080c500000000000000000000000000000000000000000000000000000000000004cc00000000000000000000000000000000000000000000000000000000000001ca00000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000bcfceff2, args=[0xb4c79dab8f259c7aee6e5b2aa729821864227e84, 0x00000000000000000000000000000000000080c5, 1228, 458, 0x00000000000000000000000000000000000000000000000000000000bcfceff2]] test_finalizeWithdrawalTransaction_differential(address,address,uint256,uint256,bytes) (runs: 29, μ: 264541, ~: 265729) 
```

🤔

**Comment by smartcontracts** [comment]
> ```
> Encountered 1 failing test in contracts/test/OptimismPortal.t.sol:OptimismPortal_FinalizeWithdrawal_Test [FAIL. Reason: EvmError: Revert Counterexample: calldata=0x2147ae39000000000000000000000000b4c79dab8f259c7aee6e5b2aa729821864227e8400000000000000000000000000000000000000000000000000000000000080c500000000000000000000000000000000000000000000000000000000000004cc00000000000000000000000000000000000000000000000000000000000001ca00000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000bcfceff2, args=[0xb4c79dab8f259c7aee6e5b2aa729821864227e84, 0x00000000000000000000000000000000000080c5, 1228, 458, 0x00000000000000000000000000000000000000000000000000000000bcfceff2]] test_finalizeWithdrawalTransaction_differential(address,address,uint256,uint256,bytes) (runs: 29, μ: 264541, ~: 265729) 
> ```
> 
> thinking

Investigating....

**Comment by smartcontracts** [comment]
Same bug as before. Opening a PR to make OptimismPortal not payable.

**Review by maurelian — APPROVED** [review]

**Review by tynes — APPROVED** [review]

**Pull request merged**